app/app.go

```diff
@@ -410,8 +431,57 @@ func NewOsmosisApp(
 		ante.DefaultSigVerificationGasConsumer,
 		encodingConfig.TxConfig.SignModeHandler(),
 		app.IBCKeeper,
+		BlockSDKAnteHandlerParams{
+			mevLane:       mevLane,
+			auctionKeeper: *app.AppKeepers.AuctionKeeper,
+			txConfig:      txConfig,
+		},
+	)
+
+	// update ante-handlers on lanes
+	opt := []base.LaneOption{
+		base.WithAnteHandler(anteHandler),
+	}
+	mevLane.WithOptions(opt...)
+	defaultLane.WithOptions(opt...)
+
+	// ABCI handlers
+	// prepare proposal
+	proposalHandler := blocksdkabci.NewProposalHandler(
+		app.Logger(),
+		txConfig.TxDecoder(),
+		txConfig.TxEncoder(),
+		lanedMempool,
+	)
+
+	// we use the block-sdk's PrepareProposal logic to build blocks
+	app.SetPrepareProposal(proposalHandler.PrepareProposalHandler())
+
+	// we use a no-op ProcessProposal, this way, we accept all proposals in avoidance
+	// of liveness failures due to Prepare / Process inconsistency. In other words,
+	// this ProcessProposal always returns ACCEPT.
+	app.SetProcessProposal(baseapp.NoOpProcessProposal())
```

Comment:
just to be sure, we're no-op'ing process proposal along with block sdk integration correct? If so, what's the reason this is ok? 👀

Reply:
Do any other chains utilize a noOpProcessPropisal? Just curious from a testing perspective what the overhead of this will need to be if we are the first to do so.